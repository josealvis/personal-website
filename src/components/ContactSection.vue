<template>
  <section id="contact">
    <div class="center-box-container">
      <h1 class="centered">
        {{ visiterName ? `${visiterName}, ` : null }}Letâ€™s work together...
      </h1>
      <span id="contact-subtitle" class="centered">contact me at</span>
      <div id="social-media-bar">
        <a v-bind:href="inAccount"
          title="linkedIn profile"
          target="_blanck"><font-awesome-icon :icon="['fab', 'linkedin']" /></a>
        <a
          v-bind:href="githubAccount"
          title="github profile"
          target="_blanck"
          ><font-awesome-icon :icon="['fab', 'github']"
        /></a>
        <a v-bind:href = "`mailto: ${email}`" title="Send me a email"><font-awesome-icon :icon="['fas', 'envelope']" /></a>
      </div>
    </div>
  </section>
</template>

<script>
import { personalInfo } from "../const/config";

export default {
  name: "ContactSection",
  props: {
    visiterName: String,
  },
  data() {
    return {
      inAccount: personalInfo.inAccount,
      githubAccount: personalInfo.gitHubAccount,
      email: personalInfo.email
    };
  },
};
</script>

<style scoped>
/*Overrided */
.center-box-container {
  padding-top: 10vh;
}

#contact {
  background-color: #161616;
  color: #fefcfb;
}

.centered {
  align-self: center;
}

#contact-subtitle {
  color: #308ec5;
  position: relative;
  top: -30px;
}
#social-media-bar {
  display: flex;
  justify-content: space-around;
  padding-top: 20px;
  font-size: 5em;
}

#social-media-bar a {
  color: #fefcfb;
}

#social-media-bar a:hover {
  animation: fade-animation;
  animation-duration: 500ms;
  animation-fill-mode: forwards;
}

@keyframes fade-animation {
  100% {
    color: #308ec5;
  }
}
</style>