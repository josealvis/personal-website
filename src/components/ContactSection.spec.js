import { mount } from '@vue/test-utils'
import ContactSection from './ContactSection.vue'

it('renders correctly', () => {
  const wrapper = mount(ContactSection)
  expect(wrapper.element).toMatchSnapshot();
});

it('renders a visiter name in title', async () => {
  const wrapper = mount(ContactSection)
  const visiterName = "Robot Test"

  const titleText = wrapper.get('[data-test="contact-title"]')

  expect(titleText.text().includes(visiterName)).toBeFalsy()

  await wrapper.setProps({ visiterName: visiterName })

  expect(titleText.text().includes(visiterName)).toBeTruthy()

})

it('renders props', async () => {
  const wrapper = mount(ContactSection, {
    data() {
      return {
        inAccount: "testInAccount",
        githubAccount: "testGitAccount",
        email: "testEmail@email.com",
      };
    }
  })

  const inIcon = wrapper.get('[data-test="in-icon"]')
  const githubIcon = wrapper.get('[data-test="github-icon"]')
  const emailIcon = wrapper.get('[data-test="email-icon"]')


  expect(inIcon.attributes().href).toEqual("testInAccount")
  expect(githubIcon.attributes().href).toEqual("testGitAccount")
  expect(emailIcon.attributes().href).toEqual("mailto: testEmail@email.com")
})