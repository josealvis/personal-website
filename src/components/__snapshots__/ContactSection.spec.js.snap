// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  class="section contact-section"
  id="contact"
>
  <div
    class="center-box-container"
  >
    <h1
      class="--centered"
      data-test="contact-title"
    >
      Let’s work together... 
    </h1>
    <span
      class="contact-section__subtitle --centered"
    >
      contact me at
    </span>
    <div
      class="social-media-bar"
    >
      <a
        class="social-media-bar__link"
        data-test="in-icon"
        href="https://www.linkedin.com/in/jose-alvis-peña-bonnet-b0194820"
        target="_blanck"
        title="linkedIn profile"
      >
        <font-awesome-icon
          icon="fab,linkedin"
        />
      </a>
      <a
        class="social-media-bar__link"
        data-test="github-icon"
        href="https://github.com/josealvis"
        target="_blanck"
        title="github profile"
      >
        <font-awesome-icon
          icon="fab,github"
        />
      </a>
      <a
        class="social-media-bar__link"
        data-test="email-icon"
        href="mailto: jalvis9114@gmail.com"
        title="Send me a email"
      >
        <font-awesome-icon
          icon="fas,envelope"
        />
      </a>
    </div>
  </div>
</section>
`;
